src/components/Checklist.tsx

```diff
@@ -0,0 +1,182 @@
+import { DOMProps } from '@react-types/shared'
+import {
+  ComponentPropsWithRef, Dispatch, ReactElement, useCallback, useEffect, useMemo, useRef, useState,
+} from 'react'
+import AnimateHeight from 'react-animate-height'
+import styled from 'styled-components'
+
+import { ChecklistFooter, ChecklistFooterProps } from './ChecklistFooter'
+import { ChecklistItemInner, ChecklistItemProps } from './ChecklistItem'
+import DropdownArrowIcon from './icons/DropdownArrowIcon'
+
+const heightAnimationDuration = 666 // 666ms
+
+const Checklist = styled(ChecklistUnstyled)(({ theme }) => ({
+  background: theme.colors['fill-two'],
+  color: theme.colors.text,
+  display: 'flex',
+  flexDirection: 'column',
+  width: 480,
+
+  '> div': {
+    borderTop: theme.borders['outline-focused'],
+    borderWidth: 4,
+    borderTopRightRadius: theme.borderRadiuses.large,
+  },
+
+  '.header': {
+    ...theme.partials.text.subtitle1,
+    borderBottom: theme.borders['fill-two'],
+    padding: `${theme.spacing.medium}px ${theme.spacing.large}px`,
+    cursor: 'pointer',
+
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+
+  '.content': {
+    padding: `${theme.spacing.xsmall}px 0`,
+    borderBottom: theme.borders['fill-two'],
+    display: 'flex',
+    flexDirection: 'column',
+  },
+
+  '.finishContainer': {
+    paddingTop: `${theme.spacing.large}px`,
+    paddingBottom: `${theme.spacing.medium}px`,
+  },
+
+  '.shrink': {
+    opacity: 0,
+    visibility: 'hidden',
+    transition: 'opacity 333ms linear, max-height 333ms linear',
+  },
+
+  '.grow': {
+    transition: 'opacity 333ms linear, max-height 333ms linear',
+  },
+
+  '.arrowUp': {
+    ...theme.partials.dropdown.arrowTransition({ isOpen: true }),
+  },
+
+  '.arrowDown': {
+    ...theme.partials.dropdown.arrowTransition({ isOpen: false }),
+  },
+}))
+
+type ChecklistProps = ComponentPropsWithRef<'div'> & {
+  label: string
+  stateProps: ChecklistStateProps
+  footerChildren: ReactElement<ChecklistFooterProps> | ReactElement<ChecklistFooterProps>[]
+  completeChildren: ReactElement
+  children: ReactElement<ChecklistItemProps>[]
+}
+
+type ChecklistStateProps = DOMProps & {
+  onSelectionChange?: Dispatch<number>
+  selectedKey?: number,
+  completedKey?: number,
+  isOpen?: boolean
+  isDismissed?: boolean
+}
```

Comment:
What is  `DOMProps` for? It doesn't seem like you're using any other props than the ones you've explicitly specified.

Reply:
Good catch. That's just a leftover. I'll remove it.